File: submissions/kasonchan/submission3/compositions/C.scala
/**
 * File: C.scala
 * Created Date: June 6, 2014
 * Status: Completed
 * Programmer: Ka-Son Chan kason.chan@ttu.edu
 * Description: This file contains the task C operations of the task graph.
 */

import java.util.Scanner
import java.io._

object C {
  def main(args: Array[String]) = {
    taskC(args(0), args(1))
  }

  def add(x: Int)(y: Int) = x + y

  def taskC(intputFileName: String, outputFileName: String) {
    // Read input file
    val scanner = new Scanner(new File(intputFileName))
    val data = scanner.nextInt()
    scanner.close()

    // Perform addition
    val result = add(data)(1)

    // Write result to output file
    val fileWriter = new PrintWriter(outputFileName)
    fileWriter.write((result).toString)
    fileWriter.write("\n")
    fileWriter.close()
  }
}

File: submissions/kasonchan/submission3/compositions/C.c
/**
 * File: C.c
 * Created Date: June 12, 2014
 * Status: Completed
 * Programmer: Ka-Son Chan kason.chan@ttu.edu
 * Description: This file contains the task C operations of the task graph.
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Function declaration
int add(int x, int y);
void taskC(char * inputFileName, char * outputFileName);

int main(int argc, char * argv[]) {
  // Check the argument number
  if (argc != 3) {
    printf("Error - invalid number of arguments\n");
  }

  taskC(argv[1], argv[2]);

  return 0;
}

// Function definitions
int add(int x, int y) {
  return x + y;
}

void taskC(char * inputFileName, char * outputFileName) {
  int result;
  FILE * fpInput;
  FILE * fdOutput;
  fpInput = fopen(inputFileName, "r");  
  fdOutput = fopen(outputFileName, "w");

  // Read from input file
  fscanf (fpInput, "%d", &result); 

  // Perform addition
  result = add(result, 1);

  // Write output to file
  fprintf(fdOutput, "%d\n", result);
  
  // Close files
  fclose(fpInput);
  fclose(fdOutput);
}
